### PR TITLE
Fix crash with fully-qualified names in module-info.java import

### LINKED ISSUE #533: Crash with module-info import
When upgrading from 0.9.2 to 0.9.3 with JDK11, the following error occurs:

```
/Users/ben/projects/caffeine/caffeine/src/main/java/module-info.java:5: error: An unhandled exception was thrown by the Error Prone static analysis plugin.
  requires static com.google.errorprone.annotations;
                                       ^
     Please report this at https://github.com/google/error-prone/issues/new and include the following:

     error-prone version: 2.10.0
     BugPattern: NullAway
     Stack Trace:
     java.lang.RuntimeException: expression is not inside a method, lambda or initializer block!
        at com.uber.nullaway.dataflow.DataFlow.resultFor(DataFlow.java:261)
        at com.uber.nullaway.dataflow.DataFlow.resultForExpr(DataFlow.java:253)
        at com.uber.nullaway.dataflow.DataFlow.expressionDataflow(DataFlow.java:196)
        at com.uber.nullaway.dataflow.AccessPathNullnessAnalysis.getNullness(AccessPathNullnessAnalysis.java:127)
        at com.uber.nullaway.NullAway.nullnessFromDataflow(NullAway.java:1979)
        at com.uber.nullaway.NullAway.mayBeNullFieldAccess(NullAway.java:1999)
        at com.uber.nullaway.NullAway.mayBeNullExpr(NullAway.java:1932)
        at com.uber.nullaway.NullAway.matchDereference(NullAway.java:2016)
        at com.uber.nullaway.NullAway.matchMemberSelect(NullAway.java:439)
        at com.google.errorprone.scanner.ErrorProneScanner.processMatchers(ErrorProneScanner.java:449)
        at com.google.errorprone.scanner.ErrorProneScanner.visitMemberSelect(ErrorProneScanner.java:726)
        at com.google.errorprone.scanner.ErrorProneScanner.visitMemberSelect(ErrorProneScanner.java:150)
        ...
```

### PR DESCRIPTION
**Pull request by msridhar**
Fixes #533 

We had no tests before for a `module-info.java` file, so we missed this regression.

### CODE REVIEW

**Comment by msridhar** [comment]
I want to fix this in a slightly different place; will re-request review when it's done.

**Comment by msridhar** [comment]
I also locally confirmed that NullAway 0.9.3 crashes on the caffeine master branch, but with this fix it compiles fine.

File: nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java [hunk 1]
```diff
@@ -307,7 +307,12 @@ public static TreePath findEnclosingMethodOrLambdaOrInitializer(TreePath path) {
    */
   public static boolean mayBeNullFieldFromType(@Nullable Symbol symbol, Config config) {
     if (symbol == null) {
-      return true;
+      // An expression representing a field dereference should always have a non-null symbol.  A
+      // `MemberSelectTree` _can_ have a null symbol in cases where it does _not_ represent a field
+      // dereference.  E.g., in a `requires` clause in a module-info.java file, the fully-qualified
+      // package name is represented using `MemberSelectTree`s.  When the symbol is null, there is
+      // no possibility of a null dereference.
+      return false;
```
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java:315]: Is this the best place to filter these out? Or could we somehow check, before we call `mayBeNullFieldFromType(...)` if we truly have a field vs requires or any other `MemberSelectTree` using construct that isn't a method call or field dereference?

Comment (msridhar) [reply nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java:315]: Yeah, I wish we could check better to enumerate the cases where we expect the symbol may be null.  `RequiresTree` was introduced in JDK 9, but with some slightly creative code and the `TreePath` to a `MemberSelectTree` we could probably detect that case.

That said, I do not see how it is possible to have a valid field dereference or method call `MemberSelectTree` that does not have a symbol.  If there is no symbol, that means the compiler could not resolve the field / method, and the code simply shouldn't compile, unless I am missing something.  We could add some stronger assertions around that.

My only hesitation is that adding stronger assertions might cause another regression / crash where things were working before.  But I guess that's better than not understanding what is going on 🙂 Let me tinker a bit.

**Comment by msridhar** [comment]
@lazaroclapp please take another look.  Here's what I've done:

* Modified `mayBeNullExpr()` to check for a `null` symbol for the member-select case, and if it is `null`, conclude that the expression cannot be null.  We already made this assumption inside `matchMemberSelect()`, and I added more comments there explaining our reasoning.
* Changed `mayBeNullFieldFromType()` to only take a non-null `symbol` parameter and fail otherwise.

I would still rather not try to enumerate every single case where a `MemberSelectTree` can have a null symbol: I think that would be fragile, and I'm very confident that anytime the symbol is null it doesn't represent a real field dereference or method call.

File: nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java [hunk 2]
```diff
@@ -300,15 +300,15 @@ public static TreePath findEnclosingMethodOrLambdaOrInitializer(TreePath path) {
   /**
    * Check if a field might be null, based on the type.
    *
-   * @param symbol symbol for field
+   * @param symbol symbol for field; must be non-null
    * @param config NullAway config
    * @return true if based on the type, package, and name of the field, the analysis should assume
    *     the field might be null; false otherwise
+   * @throws NullPointerException if {@code symbol} is null
```
Comment (lazaroclapp) [inline nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java:307]: We really should be running NullAway on NullAway 😅 

(Brings to mind a certain saying in Spanish about a blacksmith having only wooden tools)

**Review by lazaroclapp — APPROVED** [review]
Makes sense to me. Still wish we could explicitly check and enumerate the member select cases, but that's hard while supporting JDK8 through 17 :)

**Pull request merged**